### ISSUE TITLE
[sgd] LinearDataset not found

### ISSUE THREAD

**Issue by PovelikinRostislav** [ray-project/ray#8004]
### What is the problem?

RaySGD benchmark.py imports `from ray.util.sgd.torch.examples.train_example import LinearDataset`. But Ray package doesn't include samples, thus it doesn't work for packages installed from pip.
```
$ python benchmark.py --local --no-cuda
2020-04-13 17:06:28,598 INFO resource_spec.py:204 -- Starting Ray with 120.17 GiB memory available for workers and up to 55.5 GiB for objects. You can adjust these settings with ray.init(mem
ory=<bytes>, object_store_memory=<bytes>).
2020-04-13 17:06:28,946 INFO services.py:1146 -- View the Ray dashboard at localhost:8265
Traceback (most recent call last):
  File "benchmark.py", line 92, in <module>
    from ray.util.sgd.torch.examples.train_example import LinearDataset
ModuleNotFoundError: No module named 'ray.util.sgd.torch.examples'
```

*Ray version and other system information (Python version, TensorFlow version, OS):*
ray0.8.4
Ubuntu 18.04
Torch 1.4.0

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):
`python benchmark.py --local --no-cuda` from `./ray/python/ray/util/sgd/torch/examples/benchmarks` directory
If we cannot run your script, we cannot fix your issue.

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://ray.readthedocs.io/en/latest/installation.html).

**Comment by richardliaw** [comment]
@PovelikinRostislav Does this also happen on the latest wheels?

**Comment by richardliaw** [comment]
(0.9.0dev0)

**Comment by PovelikinRostislav** [comment]
@richardliaw, yes.
I assume the problem is the same as for #8002 - Ray doesn't provide examples in the package, thus trying to import examples inside `benchmark.py` with ray installed from `pip` will fail anyway

**Comment by richardliaw** [comment]
Hm, this is odd because it works for me:

```
Successfully installed ray-0.9.0.dev0
WARNING: You are using pip version 19.2.3, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
╭─ ~/Research/riselab/ray/python/ray/util/sgd/torch/examples 
╰─ cd ~/miniconda3/lib/python3.7/site-packages/ray/util/sgd/torch/examples
╭─ ~/miniconda3/lib/python3.7/site-packages/ray/util/sgd/torch/examples 
╰─ ls
__pycache__                cifar_pytorch_pbt.py       mnist_cnn.pt               sgd-development.yaml
benchmarks                 dcgan.py                   raysgd_torch_signatures.py train_example.py
cifar_pytorch_example.py   example-sgd.yaml           segmentation               tune_example.py
╭─ ~/miniconda3/lib/python3.7/site-packages/ray/util/sgd/torch/examples 
╰─ cd ..
╭─ ~/miniconda3/lib/python3.7/site-packages/ray/util/sgd/torch 
╰─ ls
__init__.py                 distributed_torch_runner.py torch_runner.py
__pycache__                 examples                    torch_trainer.py
constants.py                resnet.py                   training_operator.py
╭─ ~/miniconda3/lib/python3.7/site-packages/ray/util/sgd/torch 
╰─ raydir
╭─ ~/miniconda3/lib/python3.7/site-packages/ray/util/sgd/torch 
╰─ ipython
Python 3.7.3 (default, Mar 27 2019, 16:54:48)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from ray.util.sgd.torch.examples.train_example import LinearDataset

**Comment by PovelikinRostislav** [comment]
I checked it one more time, and it still fails. Let me show my log:
```
$ ipython
Python 3.8.2 (default, Mar 26 2020, 15:53:00)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import ray
In [2]: ray.__version__
Out[2]: '0.9.0.dev0'
In [3]: from ray.util.sgd.torch.examples.train_example import LinearDataset
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-3-4be19e3f178f> in <module>
----> 1 from ray.util.sgd.torch.examples.train_example import LinearDataset

ModuleNotFoundError: No module named 'ray.util.sgd.torch.examples'
```

And the content of the corresponding folder:
```
$ ll
/rpovelik/installed/miniconda3/envs/ray/lib/python3.8/site-packages/ray/util/sgd/torch
total 108
drwxr-xr-x 3 rpovelik intelall  4096 Apr 14 20:01 .
drwxr-xr-x 5 rpovelik intelall  4096 Apr 14 20:01 ..
-rw-r--r-- 1 rpovelik intelall   313 Apr 14 20:01 constants.py
-rw-r--r-- 1 rpovelik intelall  9850 Apr 14 20:01 distributed_torch_runner.py
-rw-r--r-- 1 rpovelik intelall   536 Apr 14 20:01 __init__.py
drwxr-xr-x 2 rpovelik intelall  4096 Apr 14 20:01 __pycache__
-rw-r--r-- 1 rpovelik intelall  4212 Apr 14 20:01 resnet.py
-rw-r--r-- 1 rpovelik intelall 12296 Apr 14 20:01 torch_runner.py
-rw-r--r-- 1 rpovelik intelall 32162 Apr 14 20:01 torch_trainer.py
-rw-r--r-- 1 rpovelik intelall 16451 Apr 14 20:01 training_operator.py
```

I installed latest wheels with `pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp38-cp38-manylinux1_x86_64.whl`

By the way, is it ok that you use 3.7 python? I'm using 3.8 here

**Comment by richardliaw** [comment]
```
╭─ ~/miniconda3/envs/py38/lib/python3.8/site-packages/ray/util/sgd/torch/examples 
╰─ ls
__pycache__                cifar_pytorch_example.py   dcgan.py                   mnist_cnn.pt               segmentation               train_example.py
benchmarks                 cifar_pytorch_pbt.py       example-sgd.yaml           raysgd_torch_signatures.py sgd-development.yaml       tune_example.py
```

```
╭─ ~/miniconda3/envs/py38/lib/python3.8/site-packages/ray/util/sgd/torch 
╰─ ll
total 192
-rw-r--r--   1 rliaw  649427843   536B Apr 16 15:43 __init__.py
drwxr-xr-x   9 rliaw  649427843   288B Apr 16 15:43 __pycache__
-rw-r--r--   1 rliaw  649427843   313B Apr 16 15:43 constants.py
-rw-r--r--   1 rliaw  649427843   9.6K Apr 16 15:43 distributed_torch_runner.py
drwxr-xr-x  14 rliaw  649427843   448B Apr 16 15:43 examples
-rw-r--r--   1 rliaw  649427843   4.1K Apr 16 15:43 resnet.py
-rw-r--r--   1 rliaw  649427843    12K Apr 16 15:43 torch_runner.py
-rw-r--r--   1 rliaw  649427843    31K Apr 16 15:43 torch_trainer.py
-rw-r--r--   1 rliaw  649427843    16K Apr 16 15:43 training_operator.py
```

**Comment by richardliaw** [comment]
hmm this is on mac though; let me try this on linxu

**Comment by richardliaw** [comment]
Got it; reproduced this on 3.8 linux.

**Comment by richardliaw** [comment]
This should be fixed with #8012.

**Comment by PovelikinRostislav** [comment]
@richardliaw I'll try!
